src/lib/room.js

```diff
@@ -204,14 +206,20 @@ export const loadMessages = async () => {
 	if (messages && messages.length) {
 		const lastMessage = messages[messages.length - 1];
 		await store.setState({ lastReadMessageId: lastMessage && lastMessage._id });
+	}
+
+	if (ongoingCall && (ongoingCall.callStatus === 'accept' || ongoingCall.callStatus === 'ongoingCallInNewTab')) {
+		return;
+	}
 
-		if (lastMessage.t === constants.webrtcCallStartedMessageType || lastMessage.t === constants.jitsiCallStartedMessageType) {
-			if (lastMessage.endTs) {
-				await store.setState({ ongoingCall: { callStatus: 'ended', time: lastMessage.ts }, incomingCallAlert: null });
-				return;
-			}
-			await processCallMessage(lastMessage);
+	const latestCallMessage = getLatestCallMessage(messages);
+	if (callStatus === 'inProgress') {
+		if (!latestCallMessage) {
+			return;
 		}
+		store.setState({ ongoingCall: { callStatus: 'ongoingCallInNewTab', time: latestCallMessage.ts }, incomingCallAlert: { show: false, callProvider: latestCallMessage[0].t } });
```

Comment:
```suggestion
		store.setState({ ongoingCall: { callStatus: 'ongoingCallInNewTab', time: latestCallMessage.ts }, incomingCallAlert: { show: false, callProvider: latestCallMessage.t } });
```

Reply:
There seems to be some inconsistency here. Does `latestCallMessage` message only contains a message object or an array of messages? In your code, here it seems like it has array of messages and over [here](https://github.com/RocketChat/Rocket.Chat.Livechat/pull/629/files#diff-5c8aa1cf19e9b52055d8dd0c0c24ebff2b3ab3bc7dec5a041ab5ba59d2f13830R222) it seems it is just a message object